transport/cluster.go

```diff
@@ -0,0 +1,295 @@
+package transport
+
+import (
+	"fmt"
+	"log"
+	"net"
+	"strconv"
+	"time"
+
+	"scylla-go-driver/frame"
+	. "scylla-go-driver/frame/response"
+
+	"go.uber.org/atomic"
+)
+
+type PeerMap = map[string]*Node
+
+type eventHandler = responseHandler
+
+type refreshHandler chan struct{}
+
+type Cluster struct {
+	peers         atomic.Value // PeerMap
+	control       *Conn
+	cfg           ConnConfig
+	events        eventHandler
+	handledEvents []frame.EventType
+	knownHosts    []string
+	refresher     refreshHandler
+
+	// Yet to be used.
+	// ring        map[Token]*Node
+	// dataCenters map[string]*DataCenter
+}
+
+// type dataCenter struct {
+// 	nodes   []Node
+// 	rackCnt uint // Type?
+// }
+
+// NewCluster also creates control connection and starts handling events and refreshing topology.
+func NewCluster(cfg ConnConfig, e []frame.EventType, hosts ...string) (*Cluster, error) {
+	if len(hosts) < 1 {
+		return nil, fmt.Errorf("at least one host is required to create cluster")
+	}
+
+	p := atomic.Value{}
+	p.Store(PeerMap{})
+	k := make([]string, len(hosts))
+	copy(k, hosts)
+
+	c := &Cluster{
+		peers:         p,
+		cfg:           cfg,
+		handledEvents: e,
+		knownHosts:    k,
+	}
+
+	if err := c.NewControl(); err != nil {
+		return nil, fmt.Errorf("creating cluster: %w", err)
```

Comment:
I'd refrain from `ing` active voice in error messages.
Plain dull `create cluster: open X: bla` is shorter and reads better.
@Michal-Leszczynski this is a general comment to all code here.

Reply:
technically it should be `create control connection` or sth.